src/mlpack/methods/reinforcement_learning/environment/acrobat.hpp

```diff
@@ -0,0 +1,317 @@
+  }
+  State InitialSample() const
+  {
+    return State((arma::randu<arma::colvec>(4) - 0.5) / 5.0);
+  }
+  /**
+   *
+   * @param state The current State
+   */        
+  bool IsTerminal(const State& state) const
+  {
+    return bool (-cos(state.Theta1())-cos(state.Theta1()
+      + state.Theta2()) > 1.0);
+  }
+  /**
+   * @param state Current State
+   * @param torque Torque Applied 
+   */
+  arma::colvec Dsdt(arma::colvec state,
+               const double torque) const
+  {
+    double m1 = linkMass1;
+    double m2 = linkMass2;
+    double l1 = linkLength1;
+    double l2 = linkLength2;
+    double lc1 = linkCom1;
+    double lc2 = linkCom2;
+    double I1 = linkMoi;
+    double I2 = linkMoi;
+    double g = gravity;
+    double a = torque;
+    double theta1 = state[0];
+    double theta2 = state[1];
+    double dtheta1 = state[2];
+    double dtheta2 = state[3];
+    double d1 = m1 * pow(lc1, 2) + m2 *
+              (pow(l1, 2) + pow(lc2, 2) + 2 * l1 * lc2 * cos(theta2))
+               + I1 + I2;
+    double d2 = m2 * (pow(lc2, 2) + l1 * lc2 * cos(theta2)) + I2;
+    double phi2 = m2 * lc2 * g * cos(theta1 + theta2 - M_PI / 2.);
+
+    double phi1 = - m2 * l1 * lc2 * pow(dtheta2, 2) * sin(theta2)
+      - 2 * m2 * l1 * lc2 * dtheta2 * dtheta1 * sin(theta2)
+      + (m1 * lc1 + m2 * l1) * g * cos(theta1 - M_PI / 2)
+      + phi2;
+
+    double ddtheta2 = (a + d2 / d1 * phi1 - m2 * l1 * lc2 * pow(dtheta1, 2) *
+     sin(theta2) - phi2) / (m2 * pow(lc2, 2) + I2 - pow(d2, 2) / d1);
+
+    double ddtheta1 = -(d2 * ddtheta2 + phi1) / d1;
+    arma::colvec returnValues = {dtheta1, dtheta2, ddtheta1, ddtheta2};
+    return returnValues;
+  };
+  /**
+   * @param value scalar value to wrap
+   * @param minimum minimum range of wrap
+   * @param maximum maximum range of wrap
+   */      
+  double Wrap(double value,
+              double minimum,
+              double maximum) const
+  {
+    double diff = maximum - minimum;
+    if (value > maximum) value = value - diff;
+    else if (value < minimum) value = value + diff;
+    return value;
+  };
+  /**
+    * @param value scalar value to bound
+    * @param minimum minimum range of bound
+    * @param maximum maximum range of bound
+    */     
+  double  Bound(double value,
+                double minimum,
+                double maximum) const
+  {
+    return std::min(std::max(value, minimum), maximum);
+  };
+
+  /**
+   * @param Action action taken
+   * 0 : negative torque
+   * 1 : zero torque
+   * 2 : positive torque
+   */
+  double Torque(const Action& action) const
+  {
+    // Add noise to the Torque
+    /*
+    * Torque is action number - 1.
+    * {0,1,2} -> {-1,0,1} 
+    */
+    double torque = double(action - 1) + mlpack::math::Random(-0.1, 0.1);
+    return torque;
+  }
+
+  /**
+    * @param state_ Current State
+    * @param Action Action Taken
+    * @param nextState nextState 
+    */
+  arma::colvec Rk4(const arma::colvec state,
```

Comment:
Do you think we could use boost odeint here?

Reply:
Do you mind giving me an example. I am not sure how you see this implementation .

Reply:
I haven't really looked into it yet, but https://www.boost.org/doc/libs/master/libs/numeric/odeint/doc/html/index.html might be helpful, this is just an idea, I don't mind to go with the current implementation.